internal/server/email/sendgrid.go

```diff
@@ -0,0 +1,84 @@
+package email
+
+import (
+	"errors"
+	"os"
+
+	"github.com/sendgrid/sendgrid-go"
+	"github.com/sendgrid/sendgrid-go/helpers/mail"
+
+	"github.com/infrahq/infra/internal/logging"
+)
+
+type EmailTemplate int8
+
+const (
+	EmailTemplateAccountCreated EmailTemplate = iota
+	EmailTemplatePasswordReset
+	EmailTemplateUserInvite
+)
+
+var emailTemplateIDs = map[EmailTemplate]string{
+	EmailTemplateAccountCreated: "",
+	EmailTemplatePasswordReset:  "d-d87873d2f11b4055befbd7064cda44d6", // transactional-password-reset
+	EmailTemplateUserInvite:     "d-ac6bcb2a7f02463c8b7fc8caffc35f2d", // transactional-user-invite // TODO: Should this instead be a "granted access to x" email?
+}
+
+var (
+	AppDomain          = "https://infrahq.com"
+	FromAddress        = "noreply@infrahq.com"
+	FromName           = "Infra"
+	TestMode           = false
+	TestDataSent       = []map[string]interface{}{}
+	ErrUnknownTemplate = errors.New("unknown template")
+	ErrNotConfigured   = errors.New("email sending not configured")
+)
+
+func IsConfigured() bool {
+	key := os.Getenv("SENDGRID_API_KEY")
+	return len(key) > 0
+}
+
+func SendTemplate(name, address string, template EmailTemplate, data map[string]interface{}) error {
+	if TestMode {
+		logging.Debugf("sent email to %q: %+v\n", address, data)
+		TestDataSent = append(TestDataSent, data)
+		return nil // quietly return
+	}
+
+	key := os.Getenv("SENDGRID_API_KEY")
```

Comment:
It'd be nice for this to come from `server.Options` as well, even if we read it from this environment variable. That way all config comes from the same place, and it should make it a bit easier to test the email package without having to set environment variables.

Reply:
We don't really want the test package hitting the sendgrid api anyway.